File: src/main/java/com/example/demoimdb/model/MovieGenre.java
package com.example.demoimdb.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import javax.security.auth.Subject;

@Entity
@Table(name = "movie_genre")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class MovieGenre {
    @ManyToOne
    @JoinColumn(name = "genre_id")
    private Genre genre;
    @ManyToOne
    @JoinColumn(name = "movie_id")
    @JsonIgnore
    private Movie movie;
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
}

File: src/entities/MovieGenre.ts
import { Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
import { Genre } from './Genre';
import { Movie } from './Movie';

@Entity({ name: 'movie_genre' })
export class MovieGenre {
  @PrimaryGeneratedColumn()
  id: number;

  @ManyToOne(() => Movie, (movie) => movie.movieGenres, {
    onDelete: 'SET NULL',
  })
  movie: Movie;

  @ManyToOne(() => Genre, (genre) => genre.movieGenres)
  genre: Genre;
}
